Google.Api.Gax.Grpc/ChannelFactory.cs

```diff
@@ -0,0 +1,21 @@
+﻿/*
+ * Copyright 2019 Google LLC
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+using Grpc.Core;
+
+namespace Google.Api.Gax.Grpc
+{
+    // TODO: Make this an abstract class instead of a delegate?
+
+    /// <summary>
+    /// Creates a channel for the given endpoint, with the given credentials and options.
+    /// </summary>
+    /// <param name="endpoint">The endpoint to connect to. Must not be null.</param>
+    /// <param name="credentials">Credentials for the channel. Must not be null.</param>
+    /// <param name="options">Options for the channel. Must not be null.</param>
+    /// <returns>A channel for the given endpoint.</returns>
+    public delegate ChannelBase ChannelFactory(ServiceEndpoint endpoint, ChannelCredentials credentials, GrpcChannelOptions options);
```

Comment:
One (dirty and hacky) idea on eliminating the Grpc.Core download:
- the hypothetical Google.Api.Gax.Grpc.ManagedClientBinding" package (which provides Grpc.Net.Client binding) also has Grpc.Core dependency, but uses some ridiculous fake version of the package so that the dependency is resolved to that version. That way, a fake version of Grpc.Core is downloaded instead of the real heavyweight package (it could contain a gutted version of the Grpc.Core assembly that does nothing). This way the Grpc.Core dependency stops being a problem.

@jskeet  WDYT?

Reply:
I thought I'd suggested that a while ago and it was rejected? It's definitely an option, but it would screw over anyone who also wanted Grpc.Core for some reason.
Let's keep thinking.